pkg/provision/gce.go

```diff
@@ -116,53 +108,73 @@ func (p *GCEProvisioner) Provision(host BasicHost) (*ProvisionedHost, error) {
 	if op.Status == gceHostRunning {
 		status = ActiveStatus
 	}
-
 	return &ProvisionedHost{
 		ID:     toGCEID(host.Name, host.Additional["zone"], host.Additional["projectid"]),
 		Status: status,
 	}, nil
 }
 
 // gceFirewallExists checks if the inlets firewall rule exists or not
-func (p *GCEProvisioner) gceFirewallExists(projectID string, firewallRuleName string, controlPort string) (bool, error) {
+func (p *GCEProvisioner) gceFirewallExists(projectID string, firewallRuleName string) (bool, error) {
 	op, err := p.gceProvisioner.Firewalls.Get(projectID, firewallRuleName).Do()
 	if err != nil {
 		return false, fmt.Errorf("could not get inlets firewall rule: %v", err)
 	}
 	if op.Name == firewallRuleName {
-		for _, firewallRule := range op.Allowed {
-			for _, port := range firewallRule.Ports {
-				if port == controlPort {
-					return true, nil
-				}
-			}
-		}
+		return true, nil
 	}
 	return false, nil
 }
 
 // createInletsFirewallRule creates a firewall rule opening up the control port for inlets
-func (p *GCEProvisioner) createInletsFirewallRule(projectID string, firewallRuleName string, controlPort string) error {
-	firewallRule := &compute.Firewall{
-		Name:        firewallRuleName,
-		Description: "Firewall rule created by inlets-operator",
-		Network:     fmt.Sprintf("projects/%s/global/networks/default", projectID),
-		Allowed: []*compute.FirewallAllowed{
-			{
-				IPProtocol: "tcp",
-				Ports:      []string{controlPort},
+func (p *GCEProvisioner) createInletsFirewallRule(projectID string, firewallRuleName string, controlPort string, pro string) error {
+	var firewallRule *compute.Firewall
+	if pro == "true" {
+		firewallRule = &compute.Firewall{
+			Name:        firewallRuleName,
+			Description: "Firewall rule created by inlets-operator",
+			Network:     fmt.Sprintf("projects/%s/global/networks/default", projectID),
+			Allowed: []*compute.FirewallAllowed{
+				{
+					IPProtocol: "tcp",
+				},
 			},
-		},
-		SourceRanges: []string{"0.0.0.0/0"},
-		Direction:    "INGRESS",
-		TargetTags:   []string{"inlets"},
+			SourceRanges: []string{"0.0.0.0/0"},
+			Direction:    "INGRESS",
+			TargetTags:   []string{"inlets"},
+		}
+	} else {
+		firewallRule = &compute.Firewall{
+			Name:        firewallRuleName,
+			Description: "Firewall rule created by inlets-operator",
+			Network:     fmt.Sprintf("projects/%s/global/networks/default", projectID),
+			Allowed: []*compute.FirewallAllowed{
+				{
+					IPProtocol: "tcp",
+					Ports:      []string{controlPort},
```

Comment:
I think this is incorrect, it also needs to open port `80`

That's why I had an issue with it and @angelbarrera92 too

Reply:
But we are aleardy opening the port with the http-server tag on the exit node (every GCE project by default has the http-server flag, privided that the user has not deleted it, it works fine)
Do you want me to create one explicitly?

Reply:
@alexellis If we're not opening the port 80 then this wouldn't have worked
https://user-images.githubusercontent.com/25264581/74101600-77150a80-4b61-11ea-9e5e-f06653e68309.png

Reply:
I've added port `80` and `443` to the inlets firewall rule now. I would've added this in the beginning but have been reluctant to do so because the firewall rules in GCE have a priority number (defaults to 1000), I didn't want to create a conflict with some locked down systems where they block ingress with a much lower priority number (means higher priority) and then have inconsistent behaviour. So left this decision to the user assuming most users never delete the default firewall-rules, as we're assuming that the user might not have deleted the `default` network. If a user deletes the `default` network then that would cause problems too.

Reply:
`443` is not required because it's not proxied by default for inlets OSS, only 8080 (control) and (80) is what should be done..